includes/rest-api/class-sensei-rest-api-course-users-controller.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * File contains the Sensei_REST_API_Course_Users_Controller class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Course users actions controller class.
+ *
+ * @since x.x.x
+ */
+class Sensei_REST_API_Course_Users_Controller extends \WP_REST_Controller {
+
+	/**
+	 * Routes namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Routes prefix.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'course-users/batch';
```

Comment:
I think the base route for the controller should be `course-users` (because the controller name is Course_Users) and add the `batch` part using the `register_rest_route` function (`$this->rest_base . '/batch'`).